comp/core/workloadmeta/workloadmeta.go

```diff
@@ -57,7 +58,21 @@ type dependencies struct {
 	Params Params
 }
 
-func newWorkloadMeta(deps dependencies) Component {
+type provider struct {
```

Comment:
Checkout https://github.com/DataDog/datadog-agent/blob/main/docs/components/migrating-to-components.md#flare for how to add flare providers to the component.